### PR TITLE
73 optimize build files

### LINKED ISSUE #73: Run deploy after build CI passes
Currently, our build CI pipeline and our deploy job run simultaneously. Also, production and staging deploy jobs have separate job definitions (two yaml files). We want to:
1. Deploy after the build CI pipeline passes.
2. Refactor the two deploy jobs into one yaml file (leverage parameter to differentiate staging vs production env).

### PR DESCRIPTION
**Pull request by alexz429**
## Summary of Changes
refactored build and deploy files to be one file working off of environment variables

resolves #73 

## Steps to Reproduce
Future PR's (including this one) and releases

NOTE: releases and deployments to production are untested until 0.9.2, which will be the first real test of these functions

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -34,3 +36,60 @@ jobs:
...
+
+      - name: Build and push version image
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          cache-to: uwcsclub/codey-bot:cache
+          cache-from: uwcsclub/codey-bot:cache
+          file: docker/${{env.DEPLOYMENT_TARGET}}/Dockerfile
+          push: true
+          tags: ${{ github.event_name == release && format('uwcsclub/codey-bot:{0},uwcsclub/codey-bot:{1}' , env.RELEASE_TAG, github.sha) || format('uwcsclub/codey-bot:{0}', github.sha)}}
```
Comment (Picowchew) [inline .github/workflows/build.yml:78]: Exactly how did you want the tag(s) to look like?

Comment (alexz429) [reply .github/workflows/build.yml]: the idea is to have
`uwcsclub/codey-bot:${{ env.RELEASE_TAG }},uwcsclub/codey-bot:${{ github.sha }}`
for releases, and 
`uwcsclub/codey-bot:${{ github.sha }}`
for pushes

Comment (alexz429) [reply .github/workflows/build.yml]: simplified it a bit
for additional context, github workflows doesn't seem to support ternary; this is the existing workaround on a as of yet unclosed ticket
https://github.com/actions/runner/issues/409

**Review by Picowchew — APPROVED** [review]
LGTM!

**Review by mcpenguin — APPROVED** [review]
LGTM too!

**Pull request merged**